src/lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper.ts

```diff
@@ -16,9 +16,39 @@ import type {
   Swapper,
   UtxoTransactionExecutionProps,
 } from 'lib/swapper/types'
+import { isSome } from 'lib/utils'
 import { executeEvmTransaction } from 'lib/utils/evm'
 
 const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+const thorswapLongtailEnabled = getConfig().REACT_APP_FEATURE_THORSWAP_LONGTAIL
+
+const getSupportedAssets = async (): Promise<[AssetId[], AssetId[]]> => {
+  let supportedSellAssetIds: AssetId[] = [thorchainAssetId]
+  let supportedBuyAssetIds: AssetId[] = [thorchainAssetId]
+  const poolResponse = await thorService.get<ThornodePoolResponse[]>(
+    `${daemonUrl}/lcd/thorchain/pools`,
+  )
+
+  const longtailTokensJson = await import('./generated/generatedThorLongtailTokens.json')
+  const longtailTokens: AssetId[] = longtailTokensJson.default
+  const l1Tokens = poolResponse.isOk()
+    ? poolResponse
+        .unwrap()
+        .data.filter(pool => pool.status === 'Available')
+        .map(pool => poolAssetIdToAssetId(pool.asset))
+        .filter(isSome)
+    : []
+
+  const allTokens = thorswapLongtailEnabled ? [...longtailTokens, ...l1Tokens] : l1Tokens
+
+  allTokens.forEach(assetId => {
+    const chainId = fromAssetId(assetId).chainId as ThorChainId
+    sellSupportedChainIds[chainId] && supportedSellAssetIds.push(assetId)
+    buySupportedChainIds[chainId] && supportedBuyAssetIds.push(assetId)
+  })
+
+  return [supportedSellAssetIds, supportedBuyAssetIds]
```

Comment:
lets return an object type here so buy and sell can never be confused in the future
```suggestion
  return { supportedSellAssetIds, supportedBuyAssetIds }
```

Reply:
Great idea. Addressed in https://github.com/shapeshift/web/pull/5638/commits/27a1c79332b85efd53dde708777cf2687aec5322.